packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -91,17 +148,99 @@ class _CupertinoScrollbarState extends State<CupertinoScrollbar> with TickerProv
     return ScrollbarPainter(
       color: _kScrollbarColor,
       textDirection: _textDirection,
-      thickness: _kScrollbarThickness,
+      thickness: _thickness,
       fadeoutOpacityAnimation: _fadeoutOpacityAnimation,
       mainAxisMargin: _kScrollbarMainAxisMargin,
       crossAxisMargin: _kScrollbarCrossAxisMargin,
-      radius: _kScrollbarRadius,
+      radius: _radius,
       padding: MediaQuery.of(context).padding,
       minLength: _kScrollbarMinLength,
       minOverscrollLength: _kScrollbarMinOverscrollLength,
     );
   }
 
+  // Handle a gesture that drags the scrollbar by the given amount.
+  void _dragScrollbar(double primaryDelta) {
+    assert(widget.controller != null);
+
+    // Convert primaryDelta, the amount that the scrollbar moved, into the
+    // coordinate space of the scroll position, and scroll to that
+    // position.
+    final double scrollOffsetLocal = _painter.getTrackToScroll(primaryDelta);
+    _dragScrollbarStartY ??= widget.controller.position.pixels;
+    widget.controller.position.jumpTo(
+      _dragScrollbarStartY + scrollOffsetLocal,
+    );
+  }
+
+  void _startFadeoutTimer() {
+    _fadeoutTimer?.cancel();
+    _fadeoutTimer = Timer(_kScrollbarTimeToFade, () {
+      _fadeoutAnimationController.reverse();
+      _fadeoutTimer = null;
+    });
+  }
+
+  void assertVertical() {
+    assert(
+      widget.controller.position.axis == Axis.vertical,
+      'Scrollbar dragging is only supported for vertical scrolling. Don\'t pass the controller param to a horizontal scrollbar.',
+    );
+  }
+
+  // Long press event callbacks handle the gesture where the user long presses
+  // on the scrollbar thumb and then drags the scrollbar without releasing.
+  void _handleLongPressStart(LongPressStartDetails details) {
+    assertVertical();
+    _fadeoutTimer?.cancel();
+    _fadeoutAnimationController.forward();
+  }
+
+  void _handleLongPress() {
+    assertVertical();
+    _fadeoutTimer?.cancel();
+    _thicknessAnimationController.forward();
+  }
+
+  void _handleLongPressMoveUpdate(LongPressMoveUpdateDetails details) {
+    assertVertical();
+    _dragScrollbar(details.localOffsetFromOrigin.dy);
+  }
+
+  void _handleLongPressEnd(LongPressEndDetails details) {
+    assertVertical();
+    _startFadeoutTimer();
+    _thicknessAnimationController.reverse();
+    _dragScrollbarStartY = null;
+    final ScrollPositionWithSingleContext scrollPosition = widget.controller.position;
+    scrollPosition.goBallistic(details.velocity.pixelsPerSecond.dy);
```

Comment:
nit: I think the initial velocity shouldn't be that of the scrollbar (i.e. touch point), but the scrollview itself (which is accessible from `scrollPosition` here I think), because of the inertia?

Reply:
I couldn't figure out how to get the velocity of the scrollview, do you know how?  I think we might not expose it. [ScrollPosition docs](https://api.flutter.dev/flutter/widgets/ScrollPosition-class.html) and [a potentially related issue](https://github.com/flutter/flutter/issues/26418).

I agree that it would be nice to not have to add velocity to the long press gesture though...

Reply:
`scrollPosition.activity.velocity`? The `scrollPostion` local variable is also a `ScrollActivityDelegate` I think.

Reply:
Oh are you saying the activity is going to be an `IdleActivity`, because we're using `jumpTo` ?
```
    widget.controller.position.jumpTo(
      _dragScrollbarStartY + scrollOffsetLocal,
    );
```

Reply:
`jumpTo` forces the pixel to `value`, which doesn't seem to respect the scrollable's scrollPhysics (it may overscroll when it shouldn't).

Reply:
Yeah I tried logging it and it was an `IdleScrollActivity` with velocity 0.0.

Reply:
At some point I tried using `.drag` and faking a drag event to see how it behaved, and to me it felt the exact same as `jumpTo`.

Reply:
@LongCatIsLooong I just tried out `.drag` instead of `.jumpTo` one more time and it turned out much better.  I think it should cover all of your concerns here and that we were talking about in person.  Take a look and let me know what you think.

After I fixed the thumb size (https://github.com/flutter/flutter/pull/36887), I was able to see some jumpiness with overscrolling.  Changing to `.drag` totally fixed it though.